diff --git a/heacore/fftw/api/Makefile.am b/heacore/fftw/api/Makefile.am
index 5cb10820..70e56b81 100644
--- a/heacore/fftw/api/Makefile.am
+++ b/heacore/fftw/api/Makefile.am
@@ -4,7 +4,7 @@ AM_CFLAGS = $(STACK_ALIGN_CFLAGS)
 EXTRA_DIST = f03api.sh genf03.pl fftw3.f03.in
 
 include_HEADERS = fftw3.h fftw3.f fftw3l.f03 fftw3q.f03
-nodist_include_HEADERS = fftw3.f03
+nodist_include_HEADERS =
 noinst_LTLIBRARIES = libapi.la
 
 libapi_la_SOURCES = apiplan.c configure.c execute-dft-c2r.c		\
diff --git a/heacore/fftw/api/Makefile.in b/heacore/fftw/api/Makefile.in
index f546d567..ce2a6bf5 100644
--- a/heacore/fftw/api/Makefile.in
+++ b/heacore/fftw/api/Makefile.in
@@ -434,7 +434,7 @@ AM_CPPFLAGS = -I $(top_srcdir)
 AM_CFLAGS = $(STACK_ALIGN_CFLAGS)
 EXTRA_DIST = f03api.sh genf03.pl fftw3.f03.in
 include_HEADERS = fftw3.h fftw3.f fftw3l.f03 fftw3q.f03
-nodist_include_HEADERS = fftw3.f03
+nodist_include_HEADERS =
 noinst_LTLIBRARIES = libapi.la
 libapi_la_SOURCES = apiplan.c configure.c execute-dft-c2r.c		\
 execute-dft-r2c.c execute-dft.c execute-r2r.c execute-split-dft-c2r.c	\
