diff --git a/heacore/heasoftpy/local-build.py b/heacore/heasoftpy/local-build.py
index f00ba587..51f8f050 100644
--- a/heacore/heasoftpy/local-build.py
+++ b/heacore/heasoftpy/local-build.py
@@ -41,6 +41,6 @@ if __name__ == '__main__':
         # call 'pip install'
         subprocess.check_call([
             sys.executable,
-            "-m", "pip", "install", ".", "--no-deps", "-t", "build",
+            "-m", "pip", "install", ".", "--no-deps", "--no-build-isolation", "-t", "build",
             "--log", "pip.log"
         ] + (["-v"] if args.verbose else []))
